app/models/launch_job_task.rb

```diff
@@ -2,7 +2,19 @@ class LaunchJobTask < MqttClientTask
   after_update :post_launch_job_task, :if => proc { state == 'completed' && status == 'ok' }
 
   def post_launch_job_task
-    PostLaunchJobTaskService.new(service_options).process
+    if tower_job_successful?
+      PostLaunchJobTaskService.new(service_options).process
+    else
+      self.status = 'error'
+      self.message = output["description"] if output.present?
```

Comment:
@hsong-rh Is the tower sending a description with the error details when it fails.

Reply:
I think so, I got the error output like this:
```
  "output": {
    "id": 46653,
    "url": "/api/v2/jobs/46653/",
    "name": "Tower RHC 38 QA Sleepy Time Survey Disabled - Setup to Fail",
    "type": "job",
    "status": "failed",
    "created": "2021-02-18T12:32:18.738474Z",
    "started": "2021-02-18T12:32:19.215109Z",
    "finished": "2021-02-18T12:32:20.661293Z",
    "modified": "2021-02-18T12:32:19.120838Z",
    "artifacts": {},
    "extra_vars": "{\"user\": \"root\"}",
    "description": "Job setup to Fail since needed survey is disabled.",
    "unified_job_template": 13
  },
```